openapi/openapi/src/main/scala/endpoints4s/ujson/JsonSchemas.scala

```diff
@@ -164,6 +164,69 @@ trait JsonSchemas extends algebra.NoDocsJsonSchemas with TuplesSchemas {
     val encoder = _ => ujson.Obj()
   }
 
+  sealed trait FieldDefaultValueEncoding
+
+  /** This object contains the options for how to encode optional fields that
+    * have default values when their value is also the field's default value.
+    *
+    * The following Scala class is the candidate example. Each encoding option
+    * includes the output that will be emitted:
+    *
+    * {{{
+    * case class Book(
+    *   name: String,
+    *   availableAsEBook: Boolean = false
+    * )
+    *
+    * val book1 = Book("Complete Imaginary Works", false)
+    * val book2 = Book("History of Writing", true)
+    * }}}
+    */
+  object FieldDefaultValueEncoding {
+
+    /** Strategy where the field is always encoded, regardless of whether it is
+      * also the default value.
+      *
+      *  - faster encoding
+      *  - larger and more complicated encoded payloads
+      *
+      * Using the `Book` example above, this strategy yields the following
+      * encoded payloads:
+      *
+      * {{{
+      * { "name": "Complete Imaginary Works", "availableAsEBook": false }
+      * { "name": "History of Writing", "availableAsEBook": true }
+      * }}}
+      */
+     object Always extends FieldDefaultValueEncoding
+
+    /** Strategy where the field is skipped if its value is also the field's
+      * default value.
+      *
+      *  - slower encoding (since a potentially expensive equality check needs
+      *    to be performed)
+      *  - smaller and simpler encoded payloads
+      *
+      * Using the `Book` example above, this strategy yields the following
+      * encoded payloads:
+      *
+      * {{{
+      * { "name": "Complete Imaginary Works" }
+      * { "name": "History of Writing", "availableAsEBook": true }
+      * }}}
+      */
+     object SkipFieldsWithDefaultValues extends FieldDefaultValueEncoding
+  }
+
+  /** Override this method to customize the behaviour of encoders produced by
+    * [[optFieldWithDefault]] when encoding a field value that corresponds to
+    * the specified default value.
+    * By default, it uses [[FieldDefaultValueEncoding.Always]].
+    *
+    * @see [[FieldDefaultValueEncoding]]
+    */
+  def defaultFieldValuesEncoding: FieldDefaultValueEncoding = FieldDefaultValueEncoding.Always
```

Comment:
I am not sure that it is easy for end-users to override this method. Could you add a test with that?

Reply:
Yep, will do. I think it should be easy to override. It would be tough to override in the OpenAPI interpreter (possibly even impossible right now, unless we thread through a duplicate option to that interpreter).